release-notes/2.1/2.1.0-preview2.md

```diff
@@ -0,0 +1,194 @@
+
+// For User Assigned Managed Identity
+"Server:{serverURL}; Authentication=Active Directory Managed Identity; User Id={ObjectIdOfManagedIdentity}; Initial Catalog={db};"
+```
+
+
+### Azure Active Directory authentication using an application client ID
+This preview release provides support for passing a user-defined application client ID to the Microsoft Authentication Library, which will be used when authenticating with Azure Active Directory.
+
+The following new APIs are introduced:
+
+1. A new constructor has been introduced in ActiveDirectoryAuthenticationProvider:\
+_[Applies to all .NET Platforms (.NET Framework, .NET Core and .NET Standard)]_
+
+```csharp
+public ActiveDirectoryAuthenticationProvider(string applicationClientId)
+```
+
+Usage:
+```csharp
+string APP_CLIENT_ID = "<GUID>";
+SqlAuthenticationProvider customAuthProvider = new ActiveDirectoryAuthenticationProvider(APP_CLIENT_ID);
+SqlAuthenticationProvider.SetProvider(SqlAuthenticationMethod.ActiveDirectoryInteractive, customAuthProvider);
+
+using (SqlConnection sqlConnection = new SqlConnection("<connection_string>")
+{
+		sqlConnection.Open();
+}
+```
+
+2. A new configuration property has been introduced under `SqlAuthenticationProviderConfigurationSection` and `SqlClientAuthenticationProviderConfigurationSection`:\
+_[Applies to .NET Framework and .NET Core]_
+
+```csharp
+internal class SqlAuthenticationProviderConfigurationSection : ConfigurationSection
+{
+		...
+		[ConfigurationProperty("applicationClientId", IsRequired = false)]
+		public string ApplicationClientId => this["applicationClientId"] as string;
+}
+
+// Inheritance
+internal class SqlClientAuthenticationProviderConfigurationSection : SqlAuthenticationProviderConfigurationSection 
+{ ... }
+```
+
+Usage:
+```xml
+<configuration>
+	<configSections>
+		<section name="SqlClientAuthenticationProviders"
+						 type="Microsoft.Data.SqlClient.SqlClientAuthenticationProviderConfigurationSection, Microsoft.Data.SqlClient" />
+	</configSections>
+	<SqlClientAuthenticationProviders applicationClientId ="<GUID>" />
+</configuration>
+
+<!--or-->
+
+<configuration>
+	<configSections>
+		<section name="SqlAuthenticationProviders"
+						 type="Microsoft.Data.SqlClient.SqlAuthenticationProviderConfigurationSection, Microsoft.Data.SqlClient" />
+	</configSections>
+	<SqlAuthenticationProviders applicationClientId ="<GUID>" />
+</configuration>
+```
+
+### "Command Timeout" connection string property
+This preview release introduces the "Command Timeout" connection string property to override the default of 30 seconds. The timeout for individual commands can be overridden using the `CommandTimeout` property.
+
+Connection string examples:
+
+`"Server:{serverURL}; Initial Catalog={db}; Integrated Security=true; Command Timeout=60"`
+
+### Increased support for Always Encrypted
+This preview release extends support for Always Encrypted on the following platforms:
```

Comment:
Does "Always Encrypted" in this context mean "Always Encrypted with enclaves" (except where noted otherwise for .NET Standard 2.0)?

Reply:
Yes.

Reply:
This change is for AE v1 (without enclaves), normal KeyStoreProvider support in .NET Standard 2.0